joint_state_publisher/joint_state_publisher/joint_state_publisher

```diff
@@ -214,9 +214,14 @@ class JointStatePublisher():
                 elif name in self.dependent_joints:
                     param = self.dependent_joints[name]
                     parent = param['parent']
-                    joint = self.free_joints[parent]
                     factor = param.get('factor', 1)
                     offset = param.get('offset', 0)
+                    while parent in self.dependent_joints:
+                        param = self.dependent_joints[parent]
+                        parent = param['parent']
+                        factor *= param.get('factor', 1)
+                        # the offset is relative only to the first parent
```

Comment:
Yes, I agree with you about the last formula (which I've already computed one month ago before testing this PR). Expanded it becomes the following:

    J_k = f_k * J_k-1 + o_k
        = f_k * f_k-1 * ... * f_2 * f_1 * J_0 +
          + f_k * (f_k-1 * (... * (f_2 * o_1 + o_2) + ...) + o_k-1) + o_k

Nonetheless I chose the other implementation because of some specific cases that I tested and which seem to me a nonsense.

With the URDF I'm saying something like:

- `Jb` has to move `x` time faster than `Jc` with an offset of `y` radians w.r.t. `Jc` itself.
- `Ja` has to move `x'` time faster than `Jb` with an offset of `y'` radians w.r.t. `Jb` itself.

Now:

1. Suppose that both `x` and `x'` are `2`, i.e. twice the velocity of their parent.
2. Suppose also that `y = 0.8' and `y' = -1.6'.

Following the formula `Ja` will move four time as faster as `Jc` (pretty fine) and it will start in the same location of `Jb` (`2 * 0.8 - 1.6 = 0`). This seems to me non intuitive starting from the URDF, because it forces me to think of all the interactions within the whole recursive chain.

Now, I see that it could be a feature for someone, but I would like to be sure that you are aware of this behavior.